website/docs/language/resources/ephemeral.mdx

```diff
@@ -0,0 +1,105 @@
+refer to that resource in the same module, but an ephemeral resource's name has no meaning
+outside that module's scope.
+
+
+## Lifecycle
+
+The lifecycle of an ephemeral resource is different from resources and data sources. When Terraform provisions ephemeral resources, it performs the following steps:
+
+1. If Terraform needs to access the result of an ephemeral resource, it opens
+that ephemeral resource. For example, if Terraform opens an ephemeral resource for a Vault secret, the Vault provider obtains a lease and returns a secret. 
+
+1. If Terraform needs access to the ephemeral resource for longer than the
+remote system's enforced expiration time, Terraform asks the provider
+to renew it periodically. For example, if Terraform renews a Vault secret ephemeral resource, the Vault provider then calls Vault's lease renewal API endpoint to extend the expiration time.
+
+1. Once Terraform no longer needs an ephemeral resource, Terraform closes
+it. This happens after the providers that depend on a certain ephemeral resource
+complete all of their work for the current Terraform run phase. For example, closing a Vault secret ephemeral resource means the Vault provider explicitly ends the lease, allowing Vault to immediately revoke the associated credentials.
+
+Terraform follows these lifecycle steps for each instance of an ephemeral
+resource in a given configuration.
+
+## Dependency graph
+
+Ephemeral resources form nodes in Terraform's dependency graph, which interact similarly as resources and data sources. For example, when a resource or data source depends on an attribute of an ephemeral resource, Terraform automatically provisions the ephemeral resource first.
+
+## Configuration model
+
+Most of the arguments within the body of an `ephemeral` block are specific to the ephemeral resource you are defining. As with resources and data sources, each provider in the [Terraform Registry](https://registry.terraform.io/browse/providers) includes documentation for the ephemeral resources it supports, if any. An ephemeral resource type's documentation lists which arguments are available and how you should format your resource's values. 
+
+The following list outlines general field hierarchy, language-specific data types, and requirements in the `ephemeral` block. 
+
+- [`ephemeral`](#ephemeral): map
+  - [`attributes`](#ephemeral)
+  - [`meta-arguments`](#ephemeral)
+
+## Complete configuration
+
+An `ephemeral` block has the following form:
+
+```hcl
+ephemeral "<resource_type>" "<resource_name>" {
+  <attributes>
+  <meta-arguments>
+}
+```
+
+## Example
+
+The following example uses the `random` provider to provision an ephemeral
+resource representing a password:
+
+```hcl
+provider random {}
+
+ephemeral "random_password" "password" {
+  length           = 16
+  special          = true
+  override_special = "!#$%&*()-_=+[]{}<>:?"
+}
+```
+
+## Referencing ephemeral resources
```

Comment:
I think I agree with Radek that we should at least mention that `provider` blocks can contain references to ephemeral values implicitly. I was planning on mentioning that in the plugin framework documentation as well

Reply:
Ooooh, I thought Radek meant we should mention ephemerality on the _provider_ side—my bad 🤦‍♀️ . Here makes sense, we *should* add it here.